src/abstractions/software/vm-translator/parser/index.js

```diff
@@ -0,0 +1,358 @@
+  parseMemoryAccessCommands (className) {
+    const segmentName = this.tokenizer.nextToken()
+    if (!isSegmentName(segmentName)) {
+      throw new CommandException(
+        `invalid segment name: ${segmentName}`, this.lineNumber)
+    }
+    const segmentIndex = parseInt(this.tokenizer.nextToken(), 10)
+    if (segmentName !== SEGMENT.CONSTANT && segmentIndex < 0) {
+      throw new CommandException('negative segment index', this.lineNumber)
+    }
+    const command = new HVMCommand(this.commandType)
+    command.setArg1(segmentName)
+    command.setArg2(segmentIndex)
+    if (segmentName === SEGMENT.STATIC) {
+      command.setStringArg(className)
+    }
+    this.instructions[this.pc] = command
+  }
+
+  /**
+   * Parses HVM commands of the format `function f k`
+   */
+  parseFunction () {
+    this.currentFunction = this.tokenizer.nextToken()
+    if (this.currentFunction === 'Sys.init') {
+      this.isSysInitFound = true
+    }
+    const numberOfLocalVariables = parseInt(this.tokenizer.nextToken(), 10)
+    if (numberOfLocalVariables < 0) {
+      throw new CommandException(
+        'negative number of lcoal variables', this.lineNumber)
+    }
+    const command = new HVMCommand(this.commandType)
+    command.setArg1(this.currentFunction)
+    command.setArg2(numberOfLocalVariables)
+    this.instructions[this.pc] = command
+  }
+
+  /**
+   * Parses HVM commands of the format `call f n`
+   */
+  parseCall () {
+    const functionName = this.tokenizer.nextToken()
+    const numberOfArgs = parseInt(this.tokenizer.nextToken(), 10)
+    const command = new HVMCommand(this.commandType)
+    command.setArg1(functionName)
+    command.setArg2(numberOfArgs)
+    this.instructions[this.pc] = command
+  }
+
+  /**
+   * Parses control flow commands:
+   * - `label l`
+   * - `goto l`
+   * - `if-goto l`
+   */
+  parseControlFlowCommands () {
+    let labelName = this.tokenizer.nextToken()
+    if (this.currentFunction !== '') {
+      labelName = this.currentFunction + '$' + labelName
+    }
+    const command = new HVMCommand(this.commandType)
+    command.setArg1(labelName)
+    this.instructions[this.pc] = command
+  }
+
+  /**
+   * Parse arithmetic or return commands
+   * - `add`, `sub`, `not`, `and`, `or`, `lt`, `gt`, `eq`, `neg`
+   * - `return`
+   */
+  parseDefault () {
+    if (this.tokenizer.countTokens() !== 1) {
+      throw new CommandException(
+        `${this.commandType} command should not have an argument`,
+        this.lineNumber)
+    }
+    this.instructions[this.pc] = new HVMCommand(this.commandType)
+  }
+
+  /**
+   * Removes comments from a line
+   * Comments can be either with // or /*.
+   * The field isSlashStar holds the current /* comment state.
+   * @param {string} line the line to uncomment
+   * @returns {string} the "un-commented" version of the given line.
+   */
+  unCommentLine (line) {
+    const output = unCommentLine(line, this.isInSlashStar)
+    this.isInSlashStar = output.isInSlashStar
```

Comment:
Again, according to the `unCommentLine` function, I don't see this value ever getting changed.

Reply:
https://github.com/nandtotetris/computer-visualizer/pull/33#discussion_r445988003